routesrv/eskipbytes.go

```diff
@@ -51,7 +59,8 @@ func (e *eskipBytes) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	defer span.Finish()
 
 	if data, initialized := e.bytes(); initialized {
-		w.Write(data)
+		w.Header().Add("Etag", e.etag)
```

Comment:
There is a race condition as etag could be modified after `e.bytes()` returns.
Lets drop `(e *eskipBytes) bytes()` and manually lock and unlock the mutex.

Reply:
Or simply make `e.bytes()` return etag as well.

Reply:
Or simply make `e.bytes()` return etag as well.

Reply:
Even though the 2nd solution(`e.bytes()` return the ETag) seems more elegant, I think we will have to go with the first one for the following reason:

1. the Second solution will make ETag generation lazy i.e. at the HTTP handler level and make ETag generation dependent on the client making the request. My worry is if the client decided to call `/routes` at a very high RPS, It can cause some performance issues.
2. With your other suggestion about using `lastModified` for the eskipBytes, I'll have to introduce manual mutex locking and unlocking anyways. So, I'll not gain much using lazy ETag generation.

Reply:
> the Second solution will make ETag generation lazy i.e. at the HTTP handler level and make ETag generation dependent on the client making the request.

I meant 
```go
func (e *eskipBytes) bytes() ([]byte, string, bool) {
	e.mu.RLock()
	defer e.mu.RUnlock()
	return e.data, e.etag, e.initialized
}
```
It could also be extended to return `lastModified` timestamp.

Reply:
Sorry, I misunderstood at first. But I did make the change based on your suggestion.